src/server/services/wfs/qgswfsutils.cpp

```diff
@@ -58,7 +58,39 @@ namespace QgsWfs
     return  href;
   }
 
-  QgsFeatureRequest parseFilterElement( const QString &typeName, QDomElement &filterElem )
+  QString layerTypeName( const QgsMapLayer *layer )
+  {
+    QString name = layer->name();
+    if ( !layer->shortName().isEmpty() )
+      name = layer->shortName();
+    name = name.replace( ' ', '_' );
+    return name;
+  }
+
+  QgsVectorLayer *layerByTypeName( const QgsProject *project, const QString &typeName )
+  {
+    QStringList layerIds = QgsServerProjectUtils::wfsLayerIds( *project );
+    for ( QStringList::const_iterator it = layerIds.constBegin(); it != layerIds.constEnd(); ++it )
```

Comment:
I was thinking about something like below instead of using iterator:

```
for ( const QString &layerId : layerIds )
{
   ...
}
```

It's simpler :).